encoding_negotiator.js

```diff
@@ -0,0 +1,17 @@
+var zlib = require('zlib')
+var Negotiator = require('negotiator')
+
+/**
+ * @const
+ * whether current node version has brotli support
+ */
+var hasBrotliSupport = 'createBrotliCompress' in zlib
+
+function negotiateEncoding (req, encodings) {
+  var negotiator = new Negotiator(req)
+
+  return negotiator.encodings(encodings, hasBrotliSupport ? ['br'] : ['gzip'])[0]
```

Comment:
Why isn't it `['br', 'gzip']` with brotli support? Or just `['br'] : []`? I'm not sure why we prefer `gzip` here but not when brotli is supported? Feels like some clients might ask for gzip?

Reply:
I suppose that when there is Brotli support, it would first check if Brotli is listed as an option in the header, and if not, it would prefer gzip, right?

Reply:
The documentation for 'preferred' should be improved to clarify how it behaves when there are more than two elements in the array or in the tests themselves (that was how I guided myself)

Reply:
The README seems clear:

> Returns an array of preferred encodings ordered by priority from a list of available encodings, while prioritizing based on preferred array between same-quality encodings.

So it uses the position in the preferred array for preference. Maybe it could probably be written a little clearer, I'm open to a PR for it.